tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,156 @@
+from json import loads
+
+from ceph.waiter import WaitUntil
+from cephci.prereq import setup_private_container_registry
+from cephci.utils.configure import get_private_registry_image, setup_ssh_keys
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.containers import Container, Registry
+from utility.log import Log
+from utility.utils import get_cephci_config
+
+log = Log(__name__)
+
+PRIVATE_REG_USER = "myregistryusername"
+PRIVATE_REG_PASS = "myregistrypassword"
+PORT = "5000"
+
+
+class ClusterCustomContainerError(Exception):
+    pass
+
+
+def get_registry_details(config):
+    """Get registry and registry credentials
+    Args:
+        config (dict): Config parameters
+    """
+    build_type = "ibm" if config.get("ibm_build") else "rh"
+    con = get_cephci_config()
+    reg_cred = con.get(f"{build_type}_registry_credentials", {})
+    return reg_cred.get("registry"), reg_cred.get("username"), reg_cred.get("password")
+
+
+def run(ceph_cluster, **kw):
+    """Verify re-deploying of monitoring stack with custom images
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer node
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Define CephAdm class object
+    cephadm = CephAdm(installer)
+
+    # Get config file and node
+    config = kw.get("config")
+
+    # Get cluster nodes
+    ceph_nodes = kw.get("ceph_nodes")
+
+    # Get dashboard image
+    registry_image = config.get("dashboard_images")
+    dashboard_images = registry_image.values()
+
+    # Get registry docker image
+    docker_reg_image = config.get("registry_image").get("docker_registry_image")
+
+    # Get the rhbuild
```

Comment:
Move this check at the start

Reply:
Done, moved this at the start.

Reply:
????

Reply:
Sorry my bad, I moved the registry, reg_username, reg_password = get_registry_details(config) to start. 
I'll move this to the start.